[Camel 4.2.0] FastCamelContext failed

### Bug description

All of the tests are running into the `UnsupportedOperationException` like

```
java.lang.RuntimeException: java.lang.ExceptionInInitializerError
	at io.quarkus.test.junit.QuarkusTestExtension.throwBootFailureException(QuarkusTestExtension.java:639)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:723)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: java.lang.ExceptionInInitializerError
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:266)
	at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:251)
	at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:606)
	at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:656)
	... 1 more
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
	... 7 more
Caused by: java.lang.UnsupportedOperationException
	at org.apache.camel.quarkus.core.FastCamelContext.createTypeConverter(FastCamelContext.java:113)
	at org.apache.camel.impl.engine.DefaultCamelContextExtension.getOrCreateTypeConverter(DefaultCamelContextExtension.java:838)
	at org.apache.camel.impl.engine.AbstractCamelContext.doBuild(AbstractCamelContext.java:2102)
	at org.apache.camel.impl.engine.SimpleCamelContext.doBuild(SimpleCamelContext.java:136)
	at org.apache.camel.support.service.BaseService.build(BaseService.java:60)
	at org.apache.camel.quarkus.core.FastCamelContext.build(FastCamelContext.java:67)
	at org.apache.camel.quarkus.core.CamelContextRecorder.createContext(CamelContextRecorder.java:76)
	at io.quarkus.deployment.steps.CamelContextProcessor$context65737263.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.CamelContextProcessor$context65737263.deploy(Unknown Source)
	... 8 more
```

It looks like related to this camel [commit](https://github.com/apache/camel/commit/f2f084c9407cfc6bc890c1d31ff9b04ad718a429#diff-fbb1df8a09e10e8b27eea4ee18615247e2647cd28a4692c69540c951d37b8cfc) .

--- a/extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelContextRecorder.java
+++ b/extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelContextRecorder.java
@@ -27,6 +27,7 @@ import org.apache.camel.CamelContext;
 import org.apache.camel.ExtendedCamelContext;
 import org.apache.camel.RouteConfigurationsBuilder;
 import org.apache.camel.RoutesBuilder;
+import org.apache.camel.TypeConverter;
 import org.apache.camel.builder.LambdaRouteBuilder;
 import org.apache.camel.builder.RouteBuilder;
 import org.apache.camel.builder.endpoint.EndpointRouteBuilder;
@@ -73,6 +74,9 @@ public class CamelContextRecorder {
         TypeConverterRegistry typeConverterRegistryValue = typeConverterRegistry.getValue();
         typeConverterRegistryValue.setInjector(new FastTypeConverterInjector(context));
         context.setTypeConverterRegistry(typeConverterRegistryValue);
+        if (typeConverterRegistryValue instanceof TypeConverter) {
+            context.setTypeConverter((TypeConverter) typeConverterRegistryValue);
+        }
         context.setLoadTypeConverters(false);
 
         extendedCamelContext.addContextPlugin(ModelJAXBContextFactory.class, contextFactory.getValue());

--- a/extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/FastCamelContext.java
+++ b/extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/FastCamelContext.java
@@ -108,6 +108,11 @@ public class FastCamelContext extends DefaultCamelContext implements CatalogCame
     }
 
     @Override
+    protected boolean eagerCreateTypeConverter() {
+        return false;
+    }
+
+    @Override
     protected TypeConverter createTypeConverter() {
         // TypeConverter impls are resolved at build time
         throw new UnsupportedOperationException();

